src/Native/System.Security.Cryptography.Native/openssl.c

```diff
@@ -911,3 +915,114 @@ UpRefEvpPkey(
 
     return CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
 }
+
+// Lock used to make sure EnsureopenSslInitialized itself is thread safe
+static pthread_mutex_t g_initLock = PTHREAD_MUTEX_INITIALIZER;
+
+// Set of locks initialized for OpenSSL
+static pthread_mutex_t* g_locks = NULL;
+
+/*
+Function:
+_LockingCallback
+
+Called back by OpenSSL to lock or unlock.
+*/
+static
+void
+_LockingCallback(int mode, int n, const char* file, int line)
```

Comment:
I was following Jeremy's lead with _MakeTimeT in same file, but I think we can just mark the functions static, _MakeTimeT is not currently. I'll go in the other direction (to static without _) as it's what I have everywhere else in the shims.